runtime/cudaq/platform/default/rest/Executor.cpp

```diff
@@ -33,8 +33,17 @@ Executor::execute(std::vector<KernelExecution> &codesToExecute) {
                 response.dump());
 
     // Add the job id and the job name.
-    ids.emplace_back(serverHelper->extractJobId(response),
-                     codesToExecute[i].name);
+
+    auto task_id = serverHelper->extractJobId(response);
+
+    if (task_id.empty()) {
+      nlohmann::json tmp(job.at("tasks"));
+      serverHelper->constructGetJobPath(tmp[0]);
```

Comment:
The RESTful assumption is that the task IDs are returned when you submit. Because of our work flow of get id then get task, they're not return on submit. So have to do this munge until that is in place. I think returning the ids on submit makes a lot of sense though and can PR it into qcaas so we can remove this